test/database.test.js

```diff
@@ -0,0 +1,66 @@
+const mocha = require('mocha');
+const assert = require('assert');
+const db = require('../database');
+const dbUrl = 'mongodb://localhost/testdb';
+
+// Necessary import to do collection cleanup.
+const mongoose = require('mongoose');
+
+// If this fails, they rest of the tests would fail anyway.
+db.connect(dbUrl);
+
+describe("Testing database operations.", function () {
+    this.timeout(5000);
+
+    after((done) => {
```

Comment:
This change lets Mocha close the test gracefully.
I have ran the tests repeatedly.
There is an occasional error which errors due to the connection persisting after the test.

Reply:
Outdated change.
The testing runs correctly now.
No errors during testing.